packages/frontend/core/src/modules/cloud/services/auth.ts

```diff
@@ -190,6 +210,26 @@ export class AuthService extends Service {
     }
   }
 
+  async signInOauthToken(token: string, provider: string) {
+    try {
+      const res = await this.fetchService.fetch('/api/oauth/callback', {
+        method: 'POST',
+        body: JSON.stringify({ token, state: this.state }),
```

Comment:
I start doubting whether it's really necessary
https://github.com/desktop/desktop/blob/8bc4ec835573972cad14c13fe1a11d5dae22ce61/app/src/lib/stores/sign-in-store.ts#L550

Reply:
the desktop authentication works more like device authentication
https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authorizing-oauth-apps#device-flow

Reply:
https://github.com/desktop/desktop/blob/development/app/src/lib/api.ts#L2185

in-app secret also exists, the difference is that our secret is requested before logging in, but github's is preset